analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -129,6 +129,36 @@ public ProgramState ResetOperations()
             ? captured
             : operation;
 
+    public IOperation ResolveCaptureAndUnwrapConversion(IOperation operation)
+    {
+        var unwrapped = false;
```

Comment:
The logic doesn't make sense to me. It should be something like
```
while(operation.Kind is FlowCaptureReference or Conversion)
{
  if(operation.Kind is Conversion)
     operation = operation.UnwrapConversion
  else if CaptureOperation.ContainsKey(....to flow capture....CaptureId)
    operation = ResolveCapture
  else
    return operation
}
return operation
```

Reply:
I find a neat solution using recursion. Let me know, what you think.